PinClustering/PinClustering/ViewControllers/MapViewController+MapHandlingExtension.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  MapViewController+MapHandlingExtension.swift
+//  PinClustering
+//
+//  Created by Norbert Agoston on 13/09/16.
+//  Copyright © 2016 Norbert Agoston. All rights reserved.
+//
+
+import Foundation
+import MapKit
+
+extension MapViewController: MKMapViewDelegate {
+    
+    func setupMap(pointsOfInterest: [AnyObject]) {
+        guard let pointOfInterest = pointsOfInterest.first as? FuelLocation else {
+            return
+        }
+        
+        centerMapOnLocation(pointOfInterest.coordinate)
+        displayOnMap(pointsOfInterest)
+    }
+    
+    func displayOnMap(pointsOfInterest: [AnyObject]) {
+        if let pointsOfInterest = pointsOfInterest as? [FuelLocation] {
+            map.addAnnotations(pointsOfInterest)
+        }
+    }
+    
+    func centerMapOnLocation(coordinate: CLLocationCoordinate2D) {
+        let regionRadius: CLLocationDistance = 50000
+        let location = CLLocation(latitude: coordinate.latitude, longitude: coordinate.longitude)
+        let coordinateRegion = MKCoordinateRegionMakeWithDistance(location.coordinate,
+                                                                  regionRadius * 2.0, regionRadius * 2.0)
+        map.setRegion(coordinateRegion, animated: true)
+    }
+    
+    func mapView(mapView: MKMapView, viewForAnnotation annotation: MKAnnotation) -> MKAnnotationView? {
+        if let annotation = annotation as? FuelLocation {
```

Comment:
a guard let instead of if let would make the code nicer. Try not to have if let in if let.